internal/parser/simple_parser_rules.go

```diff
@@ -809,3 +817,131 @@ func (p *simpleParser) parseAnalyzeStmt(r reporter) (stmt *ast.AnalyzeStmt) {
 	p.consumeToken()
 	return
 }
+
+// parseBeginStmt parses the statements as defined in the spec:
+// https://sqlite.org/lang_transaction.html
+func (p *simpleParser) parseBeginStmt(r reporter) (stmt *ast.BeginStmt) {
+	stmt = &ast.BeginStmt{}
+	p.searchNext(r, token.KeywordBegin)
+	next, ok := p.lookahead(r)
+	if !ok {
+		return
+	}
+	stmt.Begin = next
+	p.consumeToken()
+
+	next, ok = p.optionalLookahead(r)
+	if !ok || next.Type() == token.EOF {
+		return
+	}
+
+	if next.Type() == token.KeywordDeferred {
+		stmt.Deferred = next
+		p.consumeToken()
+	} else if next.Type() == token.KeywordImmediate {
+		stmt.Immediate = next
+		p.consumeToken()
+	} else if next.Type() == token.KeywordExclusive {
+		stmt.Exclusive = next
+		p.consumeToken()
+	}
+
+	next, ok = p.optionalLookahead(r)
+	if !ok || next.Type() == token.EOF {
+		return
+	}
+
+	if next.Type() == token.KeywordTransaction {
+		stmt.Transaction = next
+		p.consumeToken()
+	}
+
+	return
+}
+
+// parseCommitStmt parses the statements as defined in the spec:
+// https://sqlite.org/lang_transaction.html
+func (p *simpleParser) parseCommitStmt(r reporter) (stmt *ast.CommitStmt) {
+	stmt = &ast.CommitStmt{}
+	p.searchNext(r, token.KeywordCommit, token.KeywordEnd)
+	next, ok := p.lookahead(r)
+	if !ok {
+		return
+	}
+
+	if next.Type() == token.KeywordCommit {
+		stmt.Commit = next
+	} else if next.Type() == token.KeywordEnd {
+		stmt.End = next
+	}
+	p.consumeToken()
+
+	next, ok = p.optionalLookahead(r)
+	if !ok || next.Type() == token.EOF {
+		return
+	}
+
+	if next.Type() == token.KeywordTransaction {
+		stmt.Transaction = next
+		p.consumeToken()
+	}
+
+	return
+}
+
+// parseRollbackStmt parses the statements as defined in the spec:
+// https://sqlite.org/lang_transaction.html
+// In this function, there are a lot of cases where a KEYWORD may or may not
+// exist. We have the liberty to use "optionalLookahead" multiple times as,
```

Comment:
This does not need to be documented here, this exact behavior should be documented in the methods themselves.